File: apps/server/src/main/java/com/aistareco/aep/model/Product.java
package com.aistareco.aep.model;

import com.aistareco.common.StringListConverter;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

/**
 * Product — 商品库（v2.7；前端真值源 apps/web/src/types/product.ts）。
 * 字段名严格对齐 TypeScript {@code Product} interface。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "products")
public class Product {

    @Id
    private String id;

    @Column(nullable = false)
    private String name;

    /** 中文枚举字符串：美妆 / 食品饮料 / 数码 3C / 服饰 / 日用百货 / 母婴 / 运动 / 其他 */
    @Column(nullable = false)
    private String category;

    @Column(length = 1024)
    private String link;

    /** JSON-encoded list of image URLs。 */
    @Column(columnDefinition = "LONGTEXT")
    @Convert(converter = StringListConverter.class)
    @Builder.Default
    private List<String> images = new ArrayList<>();

    @Column(columnDefinition = "LONGTEXT")
    private String sellingPoints;

    @Column(nullable = false)
    @Builder.Default
    private int usageCount = 0;

    /** "manual" | "auto-from-generation"。 */
    @Column(nullable = false)
    @Builder.Default
    private String source = "manual";

    /** v0.26+: 商品售价（分）。可空——抖音商城短链需要解析才能拿到，老数据没有。 */
    @Column
    private Integer priceCents;

    /** v0.26+: 佣金率（0-100 整数百分比）。可空——选品表格才有此字段，URL 解析拿不到。 */
    @Column
    private Integer commissionRate;

    @Column(nullable = false)
    private LocalDate createdAt;

    @Column(nullable = false)
    private LocalDate updatedAt;
}

File: apps/web/src/types/product.ts
// ─────────────────────────────────────────────────────────────────────────────
// product.ts — 商品库领域：可被多次复用于带货视频生成的商品档案。
// 设计源：AI 明星专区 · 第三轮打磨（增加商品库）。
// ─────────────────────────────────────────────────────────────────────────────

import type { ID, ISODate } from "./_shared";

export type ProductCategory =
  | "美妆"
  | "食品饮料"
  | "数码 3C"
  | "服饰"
  | "日用百货"
  | "母婴"
  | "运动"
  | "其他";

export const PRODUCT_CATEGORIES: ProductCategory[] = [
  "美妆",
  "食品饮料",
  "数码 3C",
  "服饰",
  "日用百货",
  "母婴",
  "运动",
  "其他",
];

/** 商品来源：手动录入（用户在商品库主动创建）/ 自动落库（生成视频时由系统补建） */
export type ProductSource = "manual" | "auto-from-generation";

export interface Product {
  id: ID;
  name: string;
  category: ProductCategory;
  /** 商品外部链接（淘宝/京东/天猫/小红书等），可选 */
  link?: string;
  /** 商品图片 URL 列表（至少 1 张占位） */
  images: string[];
  /** 卖点描述（可被 AI 自动抽取覆盖） */
  sellingPoints: string;
  /** 累计被多少条带货视频引用 */
  usageCount: number;
  source: ProductSource;
  createdAt: ISODate;
  updatedAt: ISODate;
}

/** 创建 / 更新时的入参（id / 用量 / 时间戳由后端生成） */
export interface ProductInput {
  name: string;
  category: ProductCategory;
  link?: string;
  images?: string[];
  sellingPoints?: string;
  /** 默认 manual；自动落库时由后端置为 auto-from-generation */
  source?: ProductSource;
}
